tests/functional/test_awsclient.py

```diff
@@ -475,6 +491,75 @@ def test_can_provide_tags(self, stubbed_session):
             handler='app.app') == 'arn:12345:name'
         stubbed_session.verify_stubs()
 
+    def test_raises_large_deployment_error_for_connection_error(
+            self, mock_lambda_client):
+        mock_session = mock.Mock(botocore.session.Session)
```

Comment:
Offline: Going to subclass this from the botocore.stub.Stubber and add todo to port back to botocore.